### PR TITLE
SendTls13CertificateVerify: args is local ifndef WOLFSSL_ASYNC_CRYPT

### PR DESCRIPTION
**Pull request by grooverdan**
As such, don't free it.

### CODE REVIEW

**Comment by grooverdan** [comment]
Without this you get a segfault like:

<pre>
stack_bottom = 0x7f3c6551ddd8 thread_stack 0x49000
/home/dan/repos/build-mariadb-server-10.4/sql/mysqld(my_print_stacktrace+0x29)[0x55e8ccd55de9]
mysys/stacktrace.c:270(my_print_stacktrace)[0x55e8cc7d617d]
sigaction.c:0(__restore_rt)[0x7f3c6bddfc70]
:0(__GI_raise)[0x7f3c6b6bfe35]
:0(__GI_abort)[0x7f3c6b6aa895]
:0(__libc_message)[0x7f3c6b70308f]
malloc.c:0(malloc_printerr)[0x7f3c6b70a40c]
malloc.c:0(munmap_chunk)[0x7f3c6b70a6bc]
/home/dan/repos/build-mariadb-server-10.4/sql/mysqld(+0xdab497)[0x55e8ccce1497]
src/tls13.c:5456(FreeScv13Args)[0x55e8ccce5876]
src/tls13.c:8605(wolfSSL_accept_TLSv13)[0x55e8ccda8d69]
vio/viossl.c:264(ssl_handshake_loop)[0x55e8cc563acc]
sql/sql_acl.cc:13020(parse_client_handshake_packet(MPVIO_EXT*, unsigned char**, unsigned long))[0x55e8cc5642be]
sql/sql_acl.cc:13365(server_mpvio_read_packet(st_plugin_vio*, unsigned char**))[0x55e8cc553735]
sql/sql_acl.cc:14048(native_password_authenticate(st_plugin_vio*, st_mysql_server_auth_info*))[0x55e8cc55539a]
sql/sql_acl.cc:13529(do_auth_once(THD*, st_mysql_const_lex_string const*, MPVIO_EXT*))[0x55e8cc568132]
sql/sql_acl.cc:13630(acl_authenticate(THD*, unsigned int))[0x55e8cc6bf650]
sql/sql_connect.cc:1080(check_connection(THD*))[0x55e8cc6bf8a2]
sql/sql_connect.cc:1150(login_connection(THD*))[0x55e8cc6c013a]
sql/sql_connect.cc:1324(thd_prepare_connection(THD*))[0x55e8cc6c05cf]
sql/sql_connect.cc:1404(do_handle_one_connection(CONNECT*))[0x55e8cc6c0732]
</pre>

This occurs whenever a client presents a certificate to a tlsv1.3 server under the not defined `WOLFSSL_ASYNC_CRYPT` condition.

Discovered while resolving https://jira.mariadb.org/browse/MDEV-22221 / https://github.com/MariaDB/server/pull/1501

File: src/tls13.c [hunk 1]
```diff
@@ -5779,7 +5779,9 @@ static int SendTls13CertificateVerify(WOLFSSL* ssl)
 #endif /* WOLFSSL_ASYNC_CRYPT */
 
     /* Final cleanup */
+#ifdef WOLFSSL_ASYNC_CRYPT
```
Comment (dgarske) [inline src/tls13.c:5782]: Disabling this call will cause `args->sigData` to leak. I'd like to trace the root cause. What TLS 1.3 cipher suite and key type/size are you using for the MariaDB server?

Comment (grooverdan) [reply src/tls13.c:5782]: quite right on the leak. misread this.

Seems to be 8k certificates on the client/server side that doesn't occur on smaller sizes.

the server/client uses data from: https://github.com/MariaDB/server/tree/10.5/mysql-test/std_data

server:

key: server8k-key.pem
cert_file= server8k-cert.pem
ca_file=cacert.pem

client.key cert are:
client-cert.pem 
client-key.pem

client side is a gnutls based.

I tried reproducing this with
 examples/server/server -c /home/dan/repos/mariadb-server-10.4/mysql-test/std_data/server8k-cert.pem -k /home/dan/repos/mariadb-server-10.4/mysql-test/std_data/server8k-key.pem -A /home/dan/repos/mariadb-server-10.4/mysql-test/std_data/cacert.pem -i -d -v 4

however I was unable to generate the same results:

(from https://github.com/MariaDB/server/blob/10.4/mysql-test/main/ssl_8k_key.test)

appears to be RSA:

This is wolfssl from the v4.3.0-stable tag.

https://gist.github.com/grooverdan/cadb9499c2e3adc0bfbb0d22663155ff

failing that:
<pre>
git clone  --single-branch --branch 10.4-MDEV-22221-wolfssl-tls13 https://github.com/grooverdan/mariadb-server.git mariadb-server
cd mariadb-server
git submodule update --init --recursive
change mysql-test/main/ssl_8k_key.test to have TLSv1.3
mkdir  ../build
cd ../build
cmake -DPLUGIN_{ARCHIVE,TOKUDB,MROONGA,OQGRAPH,ROCKSDB,CONNECT,SPIDER}=NO   -DBUILD_CONFIG=mysql_release -DWITH_SSL=bundled  ../mariadb-server
make -j 8
mysql-test/mtr --mem {--manual-gdb} main.ssl_8k_key
</pre>

**Comment by dgarske** [comment]
Hi @grooverdan, I opened an issue #2904 to track this. Closing this PR.

**Pull request closed without merging**